server/src/test/java/io/spine/server/tuple/EitherOfThreeShould.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.Test;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * @author Alexander Yevsyukov
+ */
+@SuppressWarnings("FieldNamingConvention") // short vars are OK for tuple tests.
+public class EitherOfThreeShould {
+
+    private final StringValue a = TestValues.newUuidValue();
+    private final UInt32Value b = UInt32Value.newBuilder()
+                                             .setValue(42)
+                                             .build();
+    private final Timestamp c = Time.getCurrentTime();
+
+    private EitherOfThree<StringValue, UInt32Value, Timestamp> eitherWithA;
+    private EitherOfThree<StringValue, UInt32Value, Timestamp> eitherWithB;
+    private EitherOfThree<StringValue, UInt32Value, Timestamp> eitherWithC;
+
+    @Before
+    public void setUp() {
+        eitherWithA = EitherOfThree.withA(a);
+        eitherWithB = EitherOfThree.withB(b);
+        eitherWithC = EitherOfThree.withC(c);
+    }
+
+    @Test
+    public void support_equality() {
+        new EqualsTester().addEqualityGroup(eitherWithA, EitherOfThree.withA(a))
+                          .addEqualityGroup(eitherWithB)
+                          .addEqualityGroup(eitherWithC)
+                          .testEquals();
+    }
+
+    @Test
+    public void pass_null_tolerance_check() {
+        new NullPointerTester().testAllPublicStaticMethods(EitherOfThree.class);
+    }
+
+    @Test
+    public void return_values() {
+        assertEquals(a, eitherWithA.getA());
+        assertEquals(b, eitherWithB.getB());
+        assertEquals(c, eitherWithC.getC());
+    }
+
+    @Test
+    public void return_value_index() {
+        assertEquals(0, eitherWithA.getIndex());
+        assertEquals(1, eitherWithB.getIndex());
+        assertEquals(2, eitherWithC.getIndex());
+    }
+
+    @Test
+    public void return_only_one_value_in_iteration() {
+        final Iterator<Message> iteratorA = eitherWithA.iterator();
+
+        assertEquals(a, iteratorA.next());
+        assertFalse(iteratorA.hasNext());
+
+        final Iterator<Message> iteratorB = eitherWithB.iterator();
+
+        assertEquals(b, iteratorB.next());
+        assertFalse(iteratorA.hasNext());
+
+        final Iterator<Message> iteratorC = eitherWithC.iterator();
+
+        assertEquals(c, iteratorC.next());
+        assertFalse(iteratorA.hasNext());
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void prohibit_obtaining_the_other_value_A_B() {
+        eitherWithA.getB();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void prohibit_obtaining_the_other_value_A_C() {
+        eitherWithA.getC();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void prohibit_obtaining_the_other_value_B_A() {
+        eitherWithB.getA();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void prohibit_obtaining_the_other_value_B_C() {
```

Comment:
As long as we test both `eitherWithA` and `eitherWithB`, we should also test `eitherWithC`.

Reply:
Added. @armiol, PTAL again.